hue/hue.sh

```diff
@@ -24,127 +31,141 @@ function update_apt_get() {
   return 1
 }
 
-# Determine the role of this node
-ROLE=$(/usr/share/google/get_metadata_value attributes/dataproc-role)
-
-# Only run on the master node of the cluster
-if [[ "${ROLE}" != 'Master' ]]; then
-  exit 0
-fi
-
-# Install hue
-update_apt_get
-apt-get install -t jessie-backports hue -y
-
-# Stop hue
-systemctl stop hue
-
-cat > core-site-patch.xml <<EOF
-<property>
-  <name>hadoop.proxyuser.hue.hosts</name>
-  <value>*</value>
-</property>
-<property>
-  <name>hadoop.proxyuser.hue.groups</name>
-  <value>*</value>
-</property>
-EOF
-sed -i '/<\/configuration>/e cat core-site-patch.xml' \
-   /etc/hadoop/conf/core-site.xml
-
-
-cat > hdfs-site-patch.xml <<EOF
-<property>
-  <name>dfs.webhdfs.enabled</name>
-  <value>true</value>
-</property>
-
-<property>
-  <name>hadoop.proxyuser.hue.hosts</name>
-  <value>*</value>
-</property>
-
-<property>
-  <name>hadoop.proxyuser.hue.groups</name>
-  <value>*</value>
-</property>
+function install_hue_and_configure() {
+  local old_hdfs_url='## webhdfs_url\=http:\/\/localhost:50070'
+  local new_hdfs_url='webhdfs_url\=http:\/\/'"$(hdfs getconf -confKey  dfs.namenode.http-address)"
+  local old_mysql_settings='## engine=sqlite3(\s+)## host=(\s+)## port=(\s+)## user=(\s+)## password='
+  local new_mysql_settings='engine=mysql$1host=127.0.0.1$2port=3306$3user=hue$4password=hue-password'
+  local mysql_password='root-password'
+  local hue_password='hue-password'
+  # Install hue
+  apt-get install -t jessie-backports hue -y || err "Failed to install hue"
+
+  # Stop hue
+  systemctl stop hue || err "Hue stop action not performed"
+
+  bdconfig set_property \
+    --configuration_file 'core-site-patch.xml' \
```

Comment:
Need to pass the actual configuration file instead of a patch file to bdconfig here and below (i.e. core-site.xml instead of core-site-patch.xml), otherwise the properties don't actually get set and we get the following errors:

![hue-screenshot](https://user-images.githubusercontent.com/10858573/37931725-7215fd1e-30fb-11e8-92c9-3db58d7426a6.png)

Reply:
As a sanity check, after making revisions to this init action, it would be good to run this init action with and without the changes in this PR and do a diff of the config files to make sure they are the same.

Reply:
I compared hue.ini files for original and reviewed init action and files are identical.
I get a little bit different warnings (less) - see screenshot.  
![image](https://user-images.githubusercontent.com/25153919/37963137-a4d0eabc-31bd-11e8-8799-7b7ed75a5592.png)
```desktop.secret_key``` warning is caused by lack of secret_key value - once I filled this entry with random string warning disappeared. 

I'll try to find a fix for those warnings.
 
--